### PR TITLE
Don't write .eslintrc.json files

### PR DESCRIPTION
**Pull request by shayne-fletcher**
~Please don't merge, needs rebasing after https://github.com/digital-asset/daml/pull/5214.~

Stop writing `.eslintrc.json` files into daml2ts packages.

### CODE REVIEW

File: language-support/ts/codegen/src/TsCodeGenMain.hs [hunk 1]
```diff
@@ -632,27 +625,6 @@ writeTsConfig dir =
       , "include" .= (["src/**/*.ts"] :: [T.Text])
       ]
 
-writeEsLintConfig :: FilePath -> IO ()
```
Comment (shayne-fletcher) [inline language-support/ts/codegen/src/TsCodeGenMain.hs]: We remove this.

File: language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs [hunk 2]
```diff
@@ -218,22 +212,38 @@ tests damlTypes yarn damlc daml2ts davl = testGroup "daml2ts tests"
...
-        BSL.writeFile "package.json" $ encode (
+        BSL.writeFile "package.json" $ encode $
           object
             [ "private" .= True
             , "devDependencies" .= pkgDevDependencies configConsts
             , "workspaces" .= pkgs
             , "name" .= ("daml2ts" :: T.Text)
             , "version" .= ("0.0.0" :: T.Text)
-            ])
+            ]
```
Comment (shayne-fletcher) [inline language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs]: We write once here in the test and use it for all the packages.

**Comment by shayne-fletcher** [comment]
Sorry about this - I got into a bit of a pickle and have had to combine what was https://github.com/digital-asset/daml/pull/5214 with this.

**Comment by shayne-fletcher** [comment]
In recognition of https://github.com/digital-asset/daml/pull/5214#pullrequestreview-382201025 I have:
- Moved the `writeRootPackageJson` to `DA.Test.Daml2TsUtils`;
- Removed the newtypes `NpmPackageName` and `NpmPackageVersion`.

@cocreature , @moritzkiefer-da If you would be so kind as to re-review this aspect please?

**Comment by shayne-fletcher** [comment]
Windows failure looks transient. Will re-run once this batch is done.

File: language-support/ts/codegen/tests/src/DA/Test/Daml2TsUtils.hs [hunk 3]
```diff
@@ -0,0 +1,21 @@
+-- Copyright (c) 2020 The DAML Authors. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+{-# LANGUAGE DerivingStrategies #-}
```
Comment (hurryabit) [inline language-support/ts/codegen/tests/src/DA/Test/Daml2TsUtils.hs]: Not needed anymore.
```suggestion
```

File: language-support/ts/codegen/tests/src/DA/Test/Daml2TsUtils.hs [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+-- The need for this utility comes up in at least the assistant
+-- integration and daml2ts tests.
+writeRootPackageJson :: Maybe FilePath -> [String] -> IO ()
+writeRootPackageJson dir workspaces =
+  BSL.writeFile (maybe "package.json" (</> "package.json") dir) $ encode $
+  object
+  [ "private" .= True
+  , "workspaces" .= map T.pack workspaces
+  , "resolutions" .= HMS.fromList ([("@daml/types", "file:daml-types")] :: [(T.Text, T.Text)])
+                                ]
```
Comment (hurryabit) [inline language-support/ts/codegen/tests/src/DA/Test/Daml2TsUtils.hs]:
```suggestion
  ]
```

Comment (shayne-fletcher) [reply language-support/ts/codegen/tests/src/DA/Test/Daml2TsUtils.hs]: Oops

File: language-support/ts/codegen/tests/BUILD.bazel [hunk 5]
```diff
@@ -109,6 +109,25 @@ sh_test(
     ],
 ) if not is_windows else None
 
+da_haskell_library(
```
Comment (cocreature) [inline language-support/ts/codegen/tests/BUILD.bazel]: :+1: That should avoid the issue. I’m actually wondering if the recent changes in `rules_haskell` to not pass include directories avoids this issue even on Windows. @aherrmann probably knows.

File: language-support/ts/codegen/tests/src/DA/Test/Daml2TsUtils.hs (hunk 3 above)
Comment (shayne-fletcher) [reply language-support/ts/codegen/tests/src/DA/Test/Daml2TsUtils.hs]: Heh. I must look into hlint with an eye to understanding why that didn't get flagged! :)

**Pull request merged**